main/read.c

```diff
@@ -674,47 +674,51 @@ extern void ungetcToInputFile (int c)
 		File.ungetchBuf[File.ungetchIdx++] = c;
 }
 
-static vString *iFileGetLine (void)
+static void readLine (vString *const vLine, MIO *const mio)
 {
-	char *str;
-	size_t size;
-	bool haveLine;
+	char *str = vStringValue (vLine);
+	size_t size = vStringSize (vLine);
 
-	File.line = vStringNewOrClear (File.line);
-	str = vStringValue (File.line);
-	size = vStringSize (File.line);
+	vStringClear (vLine);
```

Comment:
while it's probably actually harmless, I'd rather call `Clear()` *before* fetching the internal pointer (and size), just in case it ever changes to reset the buffer to the default size or something.

BTW, the whole reliance on a `vString` always having a size larger than 0 seems a bit weird, but well, nothing new in this patch.